Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -59,11 +47,7 @@ extension MongoCollection {
      */
     @discardableResult
     public func insertMany(_ values: [CollectionType], options: InsertManyOptions? = nil) throws -> InsertManyResult? {
-        guard !values.isEmpty else {
-            throw UserError.invalidArgumentError(message: "values cannot be empty")
-        }
-
-        let result = try self.bulkWrite(values.map { InsertOneModel($0) }, options: BulkWriteOptions(from: options))
+        let result = try self.bulkWrite(values.map { InsertOneModel($0) }, options: options)
```

Comment:
So this one looks different than the others, is it worth taking the extra effort to make all of these methods look the same? In particular I'm wondering why e.g. an `updateMany`s thrown error will not be a complete `BulkWriteError` like `insertMany`, etc etc

Reply:
so the CRUD spec (see section [here](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#basic)) actually says that `insertMany` throws a `BulkWriteException`, whereas the rest of these methods (including `updateMany`) all throw just `WriteException`s (our equivalents are `ServerError.bulkWriteError` and `ServerError.writeError`). so while it's funky it is actually in line with the specs. 

I think an `updateMany` is really considered just a single write -- there's an `UpdateManyModel`. But an `insertMany` is considered multiple writes and has to be translated to multiple `InsertOneModel`s.

@patrickfreed may have more input as he reimplemented this as a bulk write initially.

Reply:
I think what you said covers it 👍